internal/store/memdb.go

```diff
@@ -15,30 +15,110 @@
 package store
 
 import (
-	"io/ioutil"
+	"context"
+	"encoding/csv"
+	"io"
 	"log"
+	"strings"
 
+	"cloud.google.com/go/storage"
 	"github.com/datacommonsorg/mixer/internal/parser"
 	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"google.golang.org/api/iterator"
 )
 
 // MemDb holds imported data in memory.
 type MemDb struct {
 	statSeries map[string]map[string]pb.SeriesMap
 }
 
-// Load loads tmcf + csv files into memory database
-func (memDb *MemDb) Load(tmcfFile string) error {
-	tmcfBytes, err := ioutil.ReadFile(tmcfFile)
-	if err != nil {
-		return err
+// NewMemDb initialize a MemDb instance.
+func NewMemDb() *MemDb {
+	return &MemDb{
+		statSeries: map[string]map[string]pb.SeriesMap{},
 	}
-	schemaMapping, err := parser.ParseTmcf(string(tmcfBytes))
-	// TODO: implement csv parsing and actual data loading
-	log.Printf("%v", schemaMapping)
+}
+
+// LoadFromGcs loads tmcf + csv files into memory database
+func (memDb *MemDb) LoadFromGcs(ctx context.Context, bucket, prefix string) error {
+	gcsClient, err := storage.NewClient(ctx)
 	if err != nil {
 		return err
 	}
-	memDb.statSeries = map[string]map[string]pb.SeriesMap{}
+	// The bucket should contain one tmcf and multiple compatible csv files.
+	bkt := gcsClient.Bucket(bucket)
+	objectQuery := &storage.Query{Prefix: prefix}
+	var objects []string
+	it := bkt.Objects(ctx, objectQuery)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		objects = append(objects, attrs.Name)
+	}
+	// Read TMCF
+	var schemaMapping map[string]*parser.TableSchema
+	for _, object := range objects {
+		if strings.HasSuffix(object, ".tmcf") {
+			obj := bkt.Object(object)
+			r, err := obj.NewReader(ctx)
+			if err != nil {
+				return err
+			}
+			defer r.Close()
+			buf := new(strings.Builder)
+			if _, err := io.Copy(buf, r); err != nil {
+				return err
+			}
+			schemaMapping, err = parser.ParseTmcf(buf.String())
+			if err != nil {
+				return err
+			}
+			break
+		}
+	}
```

Comment:
If you have multiple TMCFs and CSVs, how do we match them up?

Reply:
One option is we could have have a directory object per import, where each directory object has one TMCF and one or more CSVs.

Reply:
The argument "gcs_object_prefix" is used to hold one import. I make it more clear in the description.

Reply:
SG, then